SAMstrict.tex

```diff
@@ -0,0 +1,398 @@
+\samstrictrule{Inconsistent FLAGs 0x1 0x8}{The 0x8 FLAG must not be set if the 0x1 FLAG is not.}
+\samstrictrule{Inconsistent FLAGs 0x1 0x20}{The 0x20 FLAG must not be set if the 0x1 FLAG is not.}
+\samstrictrule{Inconsistent FLAGs 0x1 0x40}{The 0x40 FLAG must not be set if the 0x1 FLAG is not.}
+\samstrictrule{Inconsistent FLAGs 0x1 0x80}{The 0x80 FLAG must not be set if the 0x1 FLAG is not.}
+\samstrictrule{Inconsistent FLAGs 0x2 0x4}{The 0x2 FLAG must not be set if 0x4 is set in any primary alignments in the template}
+\samstrictrule{Inconsistent FLAGs 0x4 0x8}{The 0x8 FLAG for primary record for each segment must match the 0x4 FLAG of the primary record for the next segment in the template}
+\samstrictrule{Inconsistent FLAG 0x10 0x20}{The 0x20 FLAG must match the 0x10 FLAG for the primary alignment of the next segment in the template}
+\samstrictrule{FLAG 0x20 set on unmapped read}{The 0x10 FLAG must not be set if the 0x4 FLAG is is set.}
+\samstrictrule{Duplicate first segment primary records}{Of the primary records with the same QNAME, at most one can have FLAG 0x40 set}
+\samstrictrule{Missing first segment primary record}{Of the records with the same QNAME with 0x1 FLAG, at least one record must have FLAG 0x40 set.}
+\samstrictrule{Duplicate last segment primary records}{Of the primary records with the same QNAME, at most one can have FLAG 0x80 set}
+\samstrictrule{Missing last segment primary record}{Of the records with the same QNAME with 0x1 FLAG, at least one record must have FLAG 0x80 set.}
+\samstrictrule{Multiple primary alignment records}{Each segment must have at most one record with FLAG 0x100 and 0x800 not set.}{\samrule}
+\samstrictrule{Missing primary alignment record}{Each segment must have at least one record with FLAG 0x100 and 0x800 not set.}{\samrule}
+\samstrictrule{Unknown FLAG bit set}{FLAG bits higher than 0x800 must not be set}{\samrule}
+\samstrictrule{Unmapped reads should not have FLAG 0x10 set}{Unmapped reads should be stored in the orientation in which they came off the sequencing machine and have 0x10 unset.}{\v15bestpractice}
+
+\subsection{RNAME}
+\samstrictrule{Malformed RNAME}{RNAME must conform to the regex {\tt \char92*|[!-()+-\char60\char62-\char126][!-\char126]*}}{\samrule}
+\samstrictrule{RNAME not present in reference}{RNAME must be equal to the value of one of the SQ SN values defined in the header.}
+\samstrictrule{RNAME contains character not supported by VCFv4.3}{RNAME must not contain any of the following characters: {\tt \char60\char62\char91\char93\char58\char42}}{\vcf43}
+\samstrictrule{RNAME not supported by VCFv4.3}{RNAME must be not be one of {\tt DEL}, {\tt INS}, {\tt DUP}, {\tt INV}, {\tt CNV}, or {\tt BND}.}{\vcf43}
+\samstrictrule{RNAME does not match mate}{For a unmapped paired-end or mate-pair read whose mate is mapped, the unmapped read should have RNAME identical to its mate}{\v15bestpractice}
```

Comment:
It's terminology used for short-insert and long-insert libraries, but as far as the spec is concerned they're really both just paired reads.  Some mate-pair technologies were "outies" instead of "innies" for orientation, but the spec doesn't make any assertion on which is correct or not anyway.

If you wanted to generalise this it would need to be something like:  "unmapped reads should have the same RNAME of at least one mapped read, or '*' if all are unmapped".